stac_validator/validate.py

```diff
@@ -245,7 +247,9 @@ def recursive_validator(self, stac_type: str):
 
                     if self.log != "":
                         self.message.append(message)
-                    if self.recursive < 5:
+                    if (
+                        self.max_depth and self.max_depth < 5
+                    ):  # TODO this should be configurable, correct?
```

Comment:
The previous behavior was to append by default (recursive was -2).  The new behavior is not to append (max_depth is None).

Reply:
Yup that's a change, my bad. Should have been `if not self.max_depth or self.max_depth < 5:`, correct?

Reply:
Yeah, I think so.

Reply:
Fix in #197.